pkg/driver/nydus/nydus.go

```diff
@@ -399,3 +432,150 @@ func (d *Driver) getChunkDict(ctx context.Context, provider accelcontent.Provide
 
 	return &chunkDict, nil
 }
+
+// FetchRemoteCache fetch cache manifest from remote
+func (d *Driver) FetchRemoteCache(ctx context.Context, pvd accelcontent.Provider, ref string) error {
+	resolver, err := pvd.Resolver(ref)
+	if err != nil {
+		return err
+	}
+
+	// TODO: sets max concurrent downloaded layer limit by containerd.WithMaxConcurrentDownloads.
+	rc := &containerd.RemoteContext{
+		Resolver: resolver,
+	}
+	name, desc, err := rc.Resolver.Resolve(ctx, ref)
+	if err != nil {
+		if errors.Is(err, containerdErrDefs.ErrNotFound) {
+			// Remote cache may do not exist, just return nil
+			return nil
+		}
+		return err
+	}
+	fetcher, err := rc.Resolver.Fetcher(ctx, name)
+	if err != nil {
+		return err
+	}
+	ir, err := fetcher.Fetch(ctx, desc)
+	if err != nil {
+		if errdefs.NeedsRetryWithHTTP(err) {
+			pvd.UsePlainHTTP()
+			ir, err = fetcher.Fetch(ctx, desc)
+			if err != nil {
+				return errors.Wrap(err, "try to pull remote cache")
+			}
+		} else {
+			return errors.Wrap(err, "pull remote cache")
+		}
+	}
+
+	bytes, err := io.ReadAll(ir)
+	if err != nil {
+		return errors.Wrap(err, "read remote cache to bytes")
+	}
+	manifest := ocispec.Manifest{}
+	if err = json.Unmarshal(bytes, &manifest); err != nil {
+		return err
+	}
+
+	cs := pvd.ContentStore()
+	for _, layer := range manifest.Layers {
+		if _, err := cs.Update(ctx, content.Info{
+			Digest: layer.Digest,
+			Size:   layer.Size,
+			Labels: layer.Annotations,
+		}); err != nil {
+			if errors.Is(err, containerdErrDefs.ErrNotFound) {
+				return errdefs.ErrNotSupport
+			}
+		}
+	}
+	return nil
+}
+
+// PushRemoteCache update cache manifest and push to remote
+func (d *Driver) PushRemoteCache(ctx context.Context, pvd accelcontent.Provider, targetDesc ocispec.Descriptor, ref string) error {
+	imageConfig := ocispec.ImageConfig{}
+	imageConfigDesc, imageConfigBytes, err := nydusutils.MarshalToDesc(imageConfig, ocispec.MediaTypeImageConfig)
+	if err != nil {
+		return errors.Wrap(err, "remote cache image config marshal failed")
+	}
+	configReader := bytes.NewReader(imageConfigBytes)
+	if err = content.WriteBlob(ctx, pvd.ContentStore(), ref, configReader, *imageConfigDesc); err != nil {
+		return errors.Wrap(err, "remote cache image config write blob failed")
+	}
+
+	cs := pvd.ContentStore()
+	layers := []ocispec.Descriptor{}
+	if err = cs.Walk(ctx, func(info content.Info) error {
```

Comment:
Can we ensure the walk order is satisfied with the cache FIFO？

Reply:
unit test has been added  in content_test.